lib/trino-filesystem-alluxio/src/main/java/io/trino/filesystem/alluxio/AlluxioInputStream.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.alluxio;
+
+import alluxio.client.file.URIStatus;
+import alluxio.client.file.cache.CacheManager;
+import alluxio.conf.AlluxioConfiguration;
+import com.google.common.primitives.Ints;
+import com.google.common.primitives.Longs;
+import io.trino.filesystem.Location;
+import io.trino.filesystem.TrinoInputFile;
+import io.trino.filesystem.TrinoInputStream;
+
+import java.io.EOFException;
+import java.io.IOException;
+
+import static java.lang.Integer.max;
+import static java.lang.Math.addExact;
+import static java.lang.Math.min;
+import static java.util.Objects.checkFromIndexSize;
+import static java.util.Objects.requireNonNull;
+
+public class AlluxioInputStream
+        extends TrinoInputStream
+{
+    private final TrinoInputFile inputFile;
+    private final long fileLength;
+    private final Location location;
+    private final AlluxioCacheStats statistics;
+    private final AlluxioInputHelper helper;
+
+    private TrinoInputStream externalStream;
+    private long position;
+    private boolean closed;
+
+    public AlluxioInputStream(TrinoInputFile inputFile, URIStatus status, CacheManager cacheManager, AlluxioConfiguration configuration, AlluxioCacheStats statistics)
+    {
+        this.inputFile = requireNonNull(inputFile);
+        this.fileLength = requireNonNull(status).getLength();
+        this.location = inputFile.location();
+        this.statistics = requireNonNull(statistics);
+        this.helper = new AlluxioInputHelper(inputFile.location(), status, cacheManager, configuration, statistics);
+    }
+
+    @Override
+    public int available()
+            throws IOException
+    {
+        ensureOpen();
+
+        return Ints.saturatedCast(fileLength - position);
+    }
+
+    @Override
+    public long getPosition()
+    {
+        return position;
+    }
+
+    private void ensureOpen()
+            throws IOException
+    {
+        if (closed) {
+            throw new IOException("Output stream closed: " + location);
+        }
+    }
+
+    @Override
+    public int read()
+            throws IOException
+    {
+        ensureOpen();
+
+        byte[] bytes = new byte[1];
+        int n = read(bytes, 0, 1);
+        if (n == 1) {
+            return bytes[0] & 0xff;
```

Comment:
What's that magic constant? Extract to a const and give it a meaningful name

Reply:
It treats the byte as unsigned and ensure that the returned integer is in the range 0 to 255. If not -128 would have the wrong representation.